### PR TITLE
Feature 310 add pdf base theme as part of build

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/metaform-api/issues/310

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/metaform/server/exportTheme/ExportThemeFreemarkerTemplateLoader.kt [hunk 1]
```diff
@@ -37,21 +37,24 @@ class ExportThemeFreemarkerTemplateLoader: TemplateLoader {
...
 
-    private fun findExportThemeFile(path: String?): ExportThemeFile? {
-        return exportThemeController.findExportThemeFile(path)
-    }
-
     override fun closeTemplateSource(templateSource: Any?) {
         // Template loader is id, so no need to close
     }
 
+    private fun findExportThemeFile(path: String?): ExportThemeFile? {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/exportTheme/ExportThemeFreemarkerTemplateLoader.kt:57]: Add docs

File: src/test/java/fi/metatavu/metaform/server/test/functional/AbstractTest.kt [hunk 2]
```diff
@@ -363,6 +364,30 @@ class AbstractTest {
...
 
+    /**
+     * Checks that pdf does not contain a string
+     *
+     * @param unexpected  unexpected string
+     * @param accessToken access token
+     * @param metaform    metaform
+     * @param reply       reply
+     */
+    protected fun assertPdfDownloadNotContents(unexpected: String, accessToken: String?, metaform: Metaform, reply: Reply) {
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/AbstractTest.kt:375]: assertPdfDownloadDoesNotContain?

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt [hunk 3]
```diff
@@ -613,15 +617,20 @@ class ReplyTestsIT : AbstractTest() {
...
-                metaform.title, metaform.slug, metaform.sections, metaform.filters, metaform.scripts)
-        testBuilder.systemAdmin.metaforms.updateMetaform(newMetaform.id!!, newMetaform)
-        val reply: Reply = testBuilder.test1.replies.createSimpleReply(metaform.id!!, "Test 1, Ääkköstesti ÅÅ, Правда", ReplyMode.UPDATE)
-        assertPdfDownloadContents("content", testBuilder.systemAdmin.token, metaform, reply)
+        val simple2 = testBuilder.systemAdmin.metaforms.createFromJsonFile("simple")
+        val metaformWithTheme = testBuilder.systemAdmin.metaforms.updateMetaform(simple2.id!!, simple2.copy(exportThemeId = exportTheme.id))
+
+        // export as pdf with base theme
+        val replyWithoutTheme: Reply = testBuilder.test1.replies.createSimpleReply(metaformNoTheme.id!!, "Test 1, Ääkköstesti ÅÅ, Правда", ReplyMode.UPDATE)
+        assertPdfDownloadNotContents("content", testBuilder.systemAdmin.token, metaformNoTheme, replyWithoutTheme)
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt:629]: Why does it check for text "content" ?

**Review by anttileppa — APPROVED** [review]

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt (hunk 3 above)
Comment (kaeline) [reply src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt]: Just to show that "content" from the other theme did not get applied here unlike it did several lines lower. I added a comment about it.

File: src/test/java/fi/metatavu/metaform/server/test/functional/AbstractTest.kt (hunk 2 above)
Comment (kaeline) [reply src/test/java/fi/metatavu/metaform/server/test/functional/AbstractTest.kt]: fixed

File: src/main/kotlin/fi/metatavu/metaform/server/exportTheme/ExportThemeFreemarkerTemplateLoader.kt (hunk 1 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/metaform/server/exportTheme/ExportThemeFreemarkerTemplateLoader.kt:63]: fixed

**Pull request merged**